gcc/rust/util/rust-unicode.h

```diff
@@ -21,28 +21,44 @@
 
 #include "optional.h"
 #include "rust-system.h"
-#include "rust-lex.h"
+#include "rust-input-source.h"
 
 namespace Rust {
 
 class Utf8String
 {
 private:
-  tl::optional<std::vector<Codepoint>> chars;
+  std::vector<Codepoint> chars;
 
 public:
-  Utf8String (const std::string &maybe_utf8)
+  static tl::optional<Utf8String>
+  make_utf8_string (const std::string &maybe_utf8)
```

Comment:
This is a factory function for `Utf8String`. 
It returns an optional type value. Returns non-null value if a give `std::string` is properly encoded as UTF-8.